src/main/kotlin/no/nav/familie/ef/sak/vilkår/regler/vilkår/AleneomsorgRegel.kt

```diff
@@ -58,13 +58,22 @@ class AleneomsorgRegel(
         }
     }
 
-    private fun erDigitalSøknadOgDonorBarn(
+    private fun erDigitalSøknadOgDonorbarnSomBorMedSøker(
         metadata: HovedregelMetadata,
         barnId: UUID?,
-    ) = metadata.behandling.årsak == BehandlingÅrsak.SØKNAD &&
-        metadata.vilkårgrunnlagDto.barnMedSamvær.find {
-            it.barnId == barnId
-        }?.søknadsgrunnlag?.ikkeOppgittAnnenForelderBegrunnelse?.lowercase() == "donor"
+    ) = (metadata.behandling.årsak == BehandlingÅrsak.SØKNAD) &&
+        (!barnetHarIkkeSammeAdresseSomSøker(metadata, barnId)) && (
+            metadata.vilkårgrunnlagDto.barnMedSamvær.find {
+                it.barnId == barnId
+            }?.søknadsgrunnlag?.ikkeOppgittAnnenForelderBegrunnelse?.lowercase() == "donor"
+        )
+
+    private fun barnetHarIkkeSammeAdresseSomSøker(
+        metadata: HovedregelMetadata,
+        barnId: UUID?,
+    ) = metadata.vilkårgrunnlagDto.barnMedSamvær.find {
+        it.barnId == barnId
+    }?.søknadsgrunnlag?.skalBoBorHosSøker?.lowercase() == "nei"
```

Comment:
Hvorfor sjekker vi for NEI og ikke "JA" her?

Reply:
Kanskje fordi det er to typer "Ja"-svar?

Reply:
![image](https://github.com/navikt/familie-ef-sak/assets/141132903/c391c945-db97-43d2-ae09-5ef762b5884f)
Spørsmålet ser slik ut. Det er altså to alternativer med ja. Kunne f.eks. sjekket med contains "ja", men tenker at det ikke er like sikkert i fremtiden.